### PR TITLE
Update `tsconfig.json` to use use node16 module resolution

### PR DESCRIPTION
**Pull request by ChristianMurphy**
### Initial checklist

* [x] I read the support docs 
* [x] I read the contributing guide 
* [x] I agree to follow the code of conduct 
* [x] I searched issues and couldn’t find anything (or linked relevant results below) 
* [x] If applicable, I’ve added docs and tests

### Description of changes

Additionally scopes JSON module to the one (CJS) location where it is used.

### CODE REVIEW

**Comment by wooorm** [comment]
Where is JSON still required? 🤔

**Comment by ChristianMurphy** [comment]
> Where is JSON still required? 🤔

https://github.com/rehypejs/rehype/blob/b9e9b9949d0f37d22b0bec4ae2deb7c658521d91/packages/rehype-cli/cli.js#L8-L9

File: tsconfig.json [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
   "compilerOptions": {
-    "target": "ES2020",
-    "lib": ["ES2020"],
-    "module": "ES2020",
-    "moduleResolution": "node",
-    "resolveJsonModule": true,
+    "target": "es2021",
+    "lib": ["es2021"],
+    "module": "node16",
     "allowJs": true,
```
Comment (remcohaszing) [inline tsconfig.json:7]: Technically unrelated, but I wouldn't mind if you delete `allowJs` while you're at it. :)

Comment (ChristianMurphy) [reply tsconfig.json]: Good idea!
Removed in [c898b77](https://github.com/rehypejs/rehype/pull/123/commits/c898b7799fee0730b576d5a8e838c5c49357ab13)

**Comment by wooorm** [comment]
Ohh yeah... That's been a while. I'll probably hardcode it soon. Weird that ts doesn't understand you can do that with cjs

**Comment by ChristianMurphy** [comment]
> Weird that ts doesn't understand you can do that with cjs

Well this file technically isn't CJS, it's ESM, with a custom CJS resolve defined inside it 😅 

https://github.com/rehypejs/rehype/blob/c898b7799fee0730b576d5a8e838c5c49357ab13/packages/rehype-cli/cli.js#L6

**Comment by remcohaszing** [comment]
It kind of understands. With `"module": "node16"`, the following TypeScript:

```ts
import foo = require('bar')
```

gets compiled to:

```js
import { createModule } from 'node:module'

const require = createModule(import.meta.url)

const foo = require('bar')
```

But indeed, it doesn’t understand if this is written manually, whether in JS or TS..

**Comment by remcohaszing** [comment]
Still, I think `resolveJsonModule` isn’t needed to be able to require JSON files like that?

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
> Well this file technically isn't CJS, it's ESM, with a custom CJS resolve defined inside it 😅


It knows that it is a require call right, as it resolves things from it. Presumably it can differentiate between imports and requires. So I don’t understand why it treats it as if it would use “ESM”/import semantics instead of “CJS”/require ones

**Comment by wooorm** [comment]
Thanks!

**Comment by remcohaszing** [comment]
> I agree it likely _shouldn't_ be required.
> But it currently is.

I really thought it was required, but shouldn’t be! Anyway, I have no further comments if it is.

**Pull request merged**